commons/zenoh-shm/src/api/provider/types.rs

```diff
@@ -113,16 +117,16 @@ impl Display for MemoryLayout {
 impl MemoryLayout {
     /// Try to create a new memory layout
     #[zenoh_macros::unstable_doc]
-    pub fn new(size: usize, alignment: AllocAlignment) -> ZResult<Self> {
+    pub fn new(size: NonZeroUsize, alignment: AllocAlignment) -> Result<Self, ZLayoutError> {
```

Comment:
To improve ergonomics you could change the signature:
```rust
pub fn new<T>(size: T, alignment: AllocAlignment) -> Result<Self, ZLayoutError>
where
    T: TryInto<NonZeroUsize>
```

In this way user can still do:
```rust
let layout = MemoryLayout::new(100, AllocAlignment::default()).unwrap();
```
instead of:
```rust
let layout = MemoryLayout::new(100.try_into().unwrap(), AllocAlignment::default()).unwrap();
```

Reply:
There are two reasons why I propose to leave `Alignment` and `MemoryLayout` in it's current the form for a while:
1. I want to have a good benchmarking suite for SHM before doing any further opts. The integrated benchmarking, that is able to check big parts of our code, not just a small ones (that can be easily optimized by the compiler).
2. We need `NonZeroUsize's` Null Value Opt in some of our data structures that aggregate `MemoryLayout` for making some things very beautiful and effecient in zenoh-cpp. Placing `NonZeroUsize` in `MemoryLayout` is a good solution here.